ARCs/arc-0200.md

```diff
@@ -0,0 +1,232 @@
+        "desc": "The current balance of the holder of the token"
+      }
+    },
+    {
+      "name": "transfer",
+      "desc": "Transfers tokens",
+      "readonly": false,
+      "args": [
+        {
+          "type": "address",
+          "name": "to_",
+          "desc": "The destination of the transfer"
+        },
+        {
+          "type": "uint256",
+          "name": "value_",
+          "desc": "Amount of tokens to transfer"
+        }
+      ],
+      "returns": { "type": "bool", "desc": "Success" }
+    },
+    {
+      "name": "transferFrom",
+      "desc": "Transfers tokens from source to destination as approved spender",
+      "readonly": false,
+      "args": [
+        {
+          "type": "address",
+          "name": "from_",
+          "desc": "The source  of the transfer"
+        },
+        {
+          "type": "address",
+          "name": "to_",
+          "desc": "The destination of the transfer"
+        },
+        {
+          "type": "uint256",
+          "name": "value_",
+          "desc": "Amount of tokens to transfer"
+        }
+      ],
+      "returns": { "type": "bool", "desc": "Success" }
+    },
+    {
+      "name": "approve",
+      "desc": "Approve spender for a token",
+      "readonly": false,
+      "args": [
+        { "type": "address", "name": "spender_" },
+        { "type": "uint256", "name": "value_" }
+      ],
+      "returns": { "type": "bool", "desc": "Success" }
+    },
+    {
+      "name": "allowance",
+      "desc": "Returns the current allowance of the spender of the tokens of the owner",
+      "readonly": true,
+      "args": [
+        { "type": "address", "name": "owner_" },
+        { "type": "address", "name": "spender_" }
+      ],
+      "returns": { "type": "uint256", "desc": "The remaining allowance" }
+    }
+  ],
+  "events": [
+    {
+      "name": "Transfer",
+      "desc": "Transfer of tokens",
+      "args": [
+        {
+          "type": "address",
+          "name": "from_",
+          "desc": "The source of transfer of tokens"
+        },
+        {
+          "type": "address",
+          "name": "to_",
+          "desc": "The destination of transfer of tokens"
+        },
+        {
+          "type": "uint256",
+          "name": "value_",
+          "desc": "The amount of tokens transferred"
+        }
+      ]
+    },
+    {
+      "name": "Approval",
+      "desc": "Approval of tokens",
+      "args": [
+        {
+          "type": "address",
+          "name": "owner_",
+          "desc": "The owner of the tokens"
+        },
+        {
+          "type": "address",
+          "name": "spender_",
+          "desc": "The approved spender of tokens"
+        },
+        {
+          "type": "uint256",
+          "name": "value_",
+          "desc": "The amount of tokens approve"
+        }
+      ]
+    }
+  ]
+}
+```
+
+Ownership of a token by a zero address indicates that a token is out of circulation indefinitely, or otherwise burned or destroyed.
```

Comment:
Is there a requirement that `totalSupply` reflect this convention?  Should `totalSupply` decrease after a transfer to the ZeroAddress?

Reply:
Currently no. In my mind, `totalSupply` is immutable. That is total supply starts at the mint address. Later on it may find itself in the zero address. However, depending on how you look at it circulating supply may differ. Circulating supply could be:

1) Tokens outside of mint address and zero address * I like this one

2) Tokens outside of zero address

The rational here is that a total supply reflecting the convention can be computed by

totalSupply - balanceOf(zeroAddress) if needed

on and off chain